etl/src/energuide/extractor.py

```diff
@@ -95,16 +96,19 @@ def _extract_snippets(row: typing.Dict[str, typing.Any]) -> typing.Dict[str, typ
     return row
 
 
-def extract_data(input_path: str) -> typing.Iterator[typing.Optional[typing.Dict[str, typing.Any]]]:
+def extract_data(input_path: str,
+                 show_progress: bool = False) -> typing.Iterator[typing.Optional[typing.Dict[str, typing.Any]]]:
     validator = cerberus.Validator(INPUT_SCHEMA, purge_unknown=True)
-    for data in _read_csv(input_path):
+    rows = _read_csv(input_path, show_progress)
+
+    for row in rows:
```

Comment:
just curious, why did you change this to save the generator in `rows` (as opposed to just putting it all inline `for row in _read_csv(input_path, show_progress):`)?

Reply:
I just found it a little too dense to keep track of, once I also put the `tqdm` progress bar in `_read_csv`.